pkg/counter/prefixes.go

```diff
@@ -45,6 +45,44 @@ func NewPrefixLengthCounter(maxUniquePrefixes6, maxUniquePrefixes4 int) *PrefixL
 	}
 }
 
+// This is a bit ugly, but there's not a great way to define an IPNet without
+// parsing strings, etc.
+func createIPNet(ones, bits int) *net.IPNet {
+	return &net.IPNet{
+		Mask: net.CIDRMask(ones, bits),
+	}
+}
+
+type prefixTracker interface {
+	// GetMaxPrefixLengths returns the maximum number of unique prefix
+	// lengths that may be supported in the implementation.
+	GetMaxPrefixLengths(ipv6 bool) int
+}
+
+// DefaultPrefixLengthCounter creates a default prefix length counter that
+// already counts the minimum and maximum prefix lengths for IP hosts and
+// default routes (ie, /32 and /0)
+func DefaultPrefixLengthCounter(pt prefixTracker) *PrefixLengthCounter {
```

Comment:
Unless you have other plans for prefixTracker, it would be simpler to just pass in the ipv4 and ipv6 prefix lengths into this function. I guess that appeals to me because it means less code in this diff...

Reply:
Works for me! sorry for the hassle, though!